src/uwtools/drivers/driver.py

```diff
@@ -49,7 +49,17 @@ def resources(self) -> None:
         """
 
     @abstractmethod
-    def run(self) -> None:
+    def run(
+        self,
+        config_file: str,
```

Comment:
Since `config_file` is a required object attribute, we shouldn't need it as an argument here. 

`forecast_app` is not necessary at all given that an abstract Driver will not need any connection to an app to be able to be useful. Presumably there'd be some information provided elsewhere (maybe in the config file) to determine if this thing is running as part of an app so that other appropriate assumptions can be made.

I think the forecast model will need to be specified before we get to this point so that we can decide which class to invoke an object from. 

And `machine` should be information included in the high-level info contained in the config file. 

I think that leaves us with `dry_run`, `log_file`, `quiet`, and `verbose`, and I have other concerns there. Should those also be handled at the object attribute level? Also, do we need both `quiet` and `verbose` flags, or a single flag that handles verbosity in general with various degrees?

I did a LOT of weeding there, but perhaps the signature for `run()` should stay with just `self`.

Reply:
Ah, interesting - so I had passed all args mirroring set_config() but several of the args we had set up originally are redundant since they should be in the config file. I had left them in as a thought to be a fail-safe to check the arg against the config to ensure user certainty, but I wasn't particularly convinced by that. I think you're probably right, keep it simple, prune the redundant parts and let the validator do any work that needs to be done - thanks!

Reply:
We haven't had the full conversation yet about what to do with logging, but absent use cases I haven't imagined, I would hope that `log_file`, `quiet`, and `verbose` would all be handled far above this class, when configuring the Python base logger for the entire `uw` invocation. I think that would be handled at the cli level -- currently in each of the modules under `uwtools.cli`, soon in the single modal-cli module. (We may still need to accept an override logger from API users: I would hope to receive that via to-be-defined API functions and use it to reconfigure the Python base logger, so that no downstream logging calls need to change.)

I'm not sure about `dry_run`, but my gut feeling is that that should be set in the constructor, too.

Reply:
Working on moving anything not covered in the config file to the constructor, which I think is just `dry_run` and `log_file` now!

Reply:
hm - I think the issue is that the constructor is taking the args, so I think I should change this to see if `Forecast` is `called_once_with` the `args` and that `FV3Forecast` is just `called_once`

Reply:
Fixed!